core/src/main/java/org/polypheny/db/algebra/core/document/DocumentModify.java

```diff
@@ -66,11 +78,19 @@ public AlgDataType deriveRowType() {
 
     @Override
     public String algCompareString() {
-        String compare = "$" + getClass().getSimpleName() + "$" + operation + "$" + input.algCompareString();
-        if ( keys != null ) {
-            compare += "$" + keys.hashCode() + "$" + updates.hashCode();
-        }
-        return compare + "$" + input.algCompareString();
+        return "$" + getClass().getSimpleName() + "$" + entity.id + "$" + entity.getLayer() + "$" + operation + "$" + input.algCompareString() + "$" + updates.hashCode() + "$" + removes.hashCode() + "$" + renames.hashCode();
```

Comment:
This alg compare string locks odd. (starting with '&' and not containing a '&').

Reply:
fixed and also adjusted all other compstrings where necessary.